crypto/pkcs7/pk7_doit.c

```diff
@@ -621,14 +621,14 @@ BIO *PKCS7_dataDecode(PKCS7 *p7, EVP_PKEY *pkey, BIO *in_bio, X509 *pcert)
                 ri = sk_PKCS7_RECIP_INFO_value(rsk, i);
                 ri->ctx = p7_ctx;
                 if (pkcs7_decrypt_rinfo(&ek, &eklen, ri, pkey,
-                        EVP_CIPHER_get_key_length(cipher)) < 0)
+                        EVP_CIPHER_get_key_length(cipher)) <= 0)
```

Comment:
This change does not look right,
evp_pkey_decrypt_alloc() fails if EVP_PKEY_decrypt() fails - which looks like this loop intentionally keeps going on decrypt errors.